public/app/plugins/datasource/cloud-monitoring/datasource.ts

```diff
@@ -179,7 +179,24 @@ export default class CloudMonitoringDatasource extends DataSourceWithBackend<
           const dataQueryResponse = toDataQueryResponse({
             data: data,
           });
-          return dataQueryResponse?.data[0]?.meta?.custom?.labels ?? {};
+          const labels = dataQueryResponse?.data
+            .map((f) => f.meta?.custom?.labels)
+            .filter((p) => !!p)
+            .reduce((acc, labels) => {
+              for (let key in labels) {
+                if (!acc[key]) {
+                  acc[key] = new Set<string>();
+                }
+                acc[key].add(labels[key]);
+              }
+              return acc;
+            }, {});
+          return Object.fromEntries(
+            Object.entries(labels).map((l: any) => {
+              l[1] = Array.from(l[1]);
+              return l;
+            })
+          );
```

Comment:
A slightly simpler solution:

 ```suggestion
          const labels = dataQueryResponse?.data
            .map((f) => f.meta?.custom?.labels)
            .filter((p) => !!p)
            .reduce((acc, labels) => {
              for (let key in labels) {
                if (!acc[key]) {
                  acc[key] = [];
                }
                if (labels[key] && !acc[key].includes(labels[key])) {
                  acc[key].push(labels[key]);
                }
              }
              return acc;
            }, {});
          return labels;
```

Reply:
(I also noticed that some labels were empty, that code removes those)

Reply:
Thanks for advice!
Yes, using includes() is simpler.
But, I don’t want to use O(n) function in loop.
In some env, I guess the number of labels is too big. (I may worry too much)
Using Set is intended.

Reply:
In our env, Cloud Monitoring record more than 1000 pod_name labels.
![Screenshot 2022-01-26 091158](https://user-images.githubusercontent.com/224552/151083050-9e1fb5bc-1ed0-4f02-b170-9b714cc40a5e.png)

I think Set is faster than includes() if list is long.
https://www.measurethat.net/Benchmarks/Show/16868/1/includes-vs-has-20220126

Reply:
I am not sure if the performance benefit goes away with the later loop but anyway it's fine